dojo/tools/checkmarx_cxflow_sast/parser.py

```diff
@@ -0,0 +1,148 @@
+        else:
+            logger.warning(f"Not supported file format ${file}")
+            return list()
+
+    def _get_findings_json(self, file, test):
+        data = json.load(file)
+        findings = []
+        additional_details = data.get("additionalDetails")
+        scan_start_date = additional_details.get("scanStartDate")
+
+        issues = data.get("xissues", [])
+
+        for issue in issues:
+            vulnerability = issue.get("vulnerability")
+            status = issue.get("vulnerabilityStatus")
+            cwe = issue.get("cwe")
+            description = issue.get("description")
+            language = issue.get("language")
+            severity = issue.get("severity")
+            link = issue.get("link")
+            filename = issue.get("filename")
+            similarity_id = issue.get("similarityId")
+
+            issue_additional_details = issue.get("additionalDetails")
+            categories = issue_additional_details.get("categories")
+            results = issue_additional_details.get("results")
+
+            map_paths = {}
+
+            for result in results:
+                # all path nodes exclude sink, source, state
+                path_keys = sorted(filter(lambda k: isinstance(k, str) and k.isnumeric(), result.keys()))
+
+                path = _Path(
+                    sink=_PathNode.from_json_object(result.get("sink")),
+                    source=_PathNode.from_json_object(result.get("source")),
+                    state=result.get("state"),
+                    paths=list([result[k] for k in path_keys])
+                )
+
+                map_paths[str(path.source.line)] = path
+
+            for detail_key in issue.get("details").keys():
+                if detail_key not in map_paths:
+                    logger.warning(f"{detail_key} not found in path, ignore")
+                else:
+                    detail = map_paths[detail_key]
+
+                    finding_detail = f"**Category:** {categories}\n"
+                    finding_detail += f"**Language:** {language}\n"
+                    finding_detail += f"**Status:** {status}\n"
+                    finding_detail += f"**Finding link:** [{link}]({link})\n"
+                    finding_detail += f"**Description:** {description}\n"
+                    finding_detail += f"**Source snippet:** `{detail.source.snippet if detail.source is not None else ''}`\n"
+                    finding_detail += f"**Sink snippet:** `{detail.sink.snippet if detail.sink is not None else ''}`\n"
+
+                    finding = Finding(
+                        title=vulnerability.replace("_", " ") + " " + detail.sink.file.split("/")[
+                            -1] if detail.sink is not None else "",
+                        cwe=int(cwe),
+                        date=dateutil.parser.parse(scan_start_date),
+                        static_finding=True,
+                        unique_id_from_tool=str(similarity_id) + str(detail_key),
```

Comment:
`unique_id_from_tool ` should be coming directly from the tool in a single field, not a combo of a different fields. I am not sure how unique the `similarityId` field is, but that would be the best field to potentially use as the `unique_id_from_tool ` field. If the `similarityId` field is not really unique after all, a hash code based deduplication algorithm of `["vuln_id_from_tool", "file_path", "line"]` would be a better fit

Reply:
I removed the `unique_id_from_tool` and changed the Duplicate algorithm to `DEDUPE_ALGO_HASH_CODE.`